### ISSUE TITLE
How to dynamically update w1 and w2？The loss seems abnormal.

### ISSUE THREAD

**Issue by Wt-Yu** [Torchlight-ljj/DSMT-Net#7]
with torch.no_grad():
        w1.data -= 0.01 * w1.grad.data  
        w2.data -= 0.01 * w2.grad.data
[15:44:41.665052] Start training for 400 epochs
[15:44:41.667174] log_dir: data_pretrain
[15:44:45.719364] Epoch: [0]  [   0/1662]  eta: 1:52:13  lr: 0.000000  loss: 6.5652 (6.5652)  time: 4.0512  data: 2.0910  max mem: 6501
[15:44:54.698398] Epoch: [0]  [  20/1662]  eta: 0:16:58  lr: 0.000000  loss: -216792896.0000 (-175468937.5802)  time: 0.4489  data: 0.0002  max mem: 6516
[15:45:04.332503] Epoch: [0]  [  40/1662]  eta: 0:14:56  lr: 0.000000  loss: -266915552.0000 (-219758741.6874)  time: 0.4817  data: 0.0002  max mem: 7381
[15:45:14.172964] Epoch: [0]  [  60/1662]  eta: 0:14:13  lr: 0.000000  loss: -339819648.0000 (-259741210.6424)  time: 0.4920  data: 0.0002  max mem: 7384
